help_to_heat/frontdoor/views.py

```diff
@@ -279,44 +359,70 @@ def get(self, request, session_id, page_name, data=None, errors=None, is_change_
         if not ("safe_to_cache" in context and context["safe_to_cache"]):
             response["cache-control"] = "no-store"
             response["Pragma"] = "no-cache"
-        return self.handle_get(response, request, session_id, page_name, context)
-
-    def handle_get(self, response, request, session_id, page_name, context):
         return response
 
-    def get_prev_next_urls(self, session_id, page_name):
-        return get_prev_next_urls(session_id, page_name)
+    def get_extra_context(self, request, session_id, page_name, data):
+        return {}
 
     def post(self, request, session_id, page_name, is_change_page=False):
         data = request.POST.dict()
         errors = self.validate(request, session_id, page_name, data, is_change_page)
         if errors:
-            return self.get(request, session_id, page_name, data=data, errors=errors, is_change_page=is_change_page)
+            return self.get(
+                request, session_id, page_name, extra_data=data, errors=errors, is_change_page=is_change_page
+            )
         else:
             try:
-                data = self.save_data(request, session_id, page_name)
+                data = self.save_post_data(data, session_id, page_name)
+                data = interface.api.session.save_answer(session_id, page_name, data)
```

Comment:
the second assignment was in the project already. I'm not immediately sure why it's done so if it looks extraneous to you too then it can be removed

Reply:
Certainly it's something I'd like to get rid of - at the very least let's assign to different variables rather than reusing the one. Likely to be quite confusing to debug otherwise.

Are these methods returning the original `data` object that gets passed in? Are they transforming it at all?

Reply:
`save_post_data` often does transform the object, but sometimes returns a new object. it depends on what's easiest for the subclass's function. if we'd like no mutations to the passed object I could copy it before passing in

looking into it further, the returned data after saving an answer isnt used any more so can be removed